src/main/java/io/axoniq/axonserver/connector/query/impl/QueryChannelImpl.java

```diff
@@ -190,25 +194,26 @@ public void complete() {
     @Override
     public synchronized void connect() {
         if (outboundQueryStream.get() != null) {
-            // we're already connected on this channel
+            logger.debug("QueryChannel for context '{}' is already connected", context);
             return;
         }
         IncomingQueryInstructionStream responseObserver = new IncomingQueryInstructionStream(
-                clientIdentification.getClientId(), permits, permitsBatch, e -> scheduleReconnect()
+                clientIdentification.getClientId(), permits, permitsBatch, e -> scheduleReconnect(),
+                upstream -> {
```

Comment:
Why use a method `registerOutboundStream` in the `CommandChannelImpl` and a lambda in this class?